# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [1]:
library(tidyverse) 

── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──
✔ ggplot2 2.2.1     ✔ purrr   0.2.4
✔ tibble  1.4.1     ✔ dplyr   0.7.4
✔ tidyr   0.7.2     ✔ stringr 1.2.0
✔ readr   1.1.1     ✔ forcats 0.2.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [2]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3         V4        V5        V6        V7       
1 0.5279673 0.4911516 0.98927904 0.4219362 0.6606145 0.3168457 0.3927465
2 0.4973708 0.1010168 0.04712431 0.4651611 0.9418145 0.2303550 0.3222190
  V8        V9        V10       V11       V12       V13       V14       
1 0.1476588 0.5662950 0.8049022 0.5866859 0.5081062 0.4960016 0.01397645
2 0.2691683 0.3042096 0.8719170 0.6696469 0.6439612 0.6483106 0.79796179
  V15       V16       V17       V18       V19        V20      
1 0.5069646 0.6271707 0.1118951 0.3945029 0.06353129 0.4794909
2 0.9411412 0.9491544 0.2120284 0.5176771 0.19456273 0.2858688

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [3]:
df2 <- df1 %>% 
    gather(Key, Value, V1:V20)
head(df2)

Key Value     
1 V1  0.52796727
2 V1  0.49737082
3 V2  0.49115163
4 V2  0.10101683
5 V3  0.98927904
6 V3  0.04712431

## Assignment 2

We will generate some data:

In [4]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [5]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [6]:
df5 <- full_join(df3, df4, by="id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook